File: 09-string/01-reverse-a-string/01-reverse-a-string.ts
const reverse_a_string = (s: string) => {
  const str = [...Array(s.length)].map(() => '')
  let [l, r] = [0, s.length - 1]
  while (l <= r) {
    str[l] = s[r]
    str[r] = s[l]
    l = l + 1
    r = r - 1
  }
  return str.join('')
}

console.log(reverse_a_string('abdul'))
console.log(reverse_a_string('hathi'))

console.log(reverse_a_string('aysha'))

File: 09-string/01-reverse-a-string/01-reverse-a-string.rb
def reverse_a_string(s)
  l, r = 0, s.length - 1
  while l < r
    s[l], s[r] = s[r], s[l]
    l, r = l+1, r-1
  end
  s
end

puts reverse_a_string 'abdul'

puts reverse_a_string 'hathi'